scripts/src/public/services/orders.ts

```diff
@@ -110,13 +112,18 @@ export async function createMarketplaceOrder(offerId: string, user: User, logger
 }
 
 export async function createExternalOrder(jwt: string, user: User, logger: LoggerInstance): Promise<OpenOrder> {
-	const offer = await validateSpendJWT(jwt, logger);
+	const payload = await validateExternalOrderJWT(jwt, logger);
+
+	if (payload.sub === "earn" && (payload as EarnPayload).user_id !== user.id) {
```

Comment:
The user_id in the EarnPayload is an `app_user_id` The Digital Service is never exposed to the internal User.id.
You should check `(payload as EarnPayload).user_id !== user.app_user_id`